src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocNotificationService.java

```diff
@@ -49,25 +57,73 @@ public class NocNotificationService {
     private String tribunalTemplateId;
 
     public void sendNotificationOfChangeEmails(CaseDetails caseDetailsPrevious,
-                                               CaseDetails caseDetailsNew,
-                                               ChangeOrganisationRequest changeRequest) {
+                             CaseDetails caseDetailsNew,
+                             ChangeOrganisationRequest changeRequest,
+                             String currentUserEmail) {
+        DynamicFixedListType caseRoleId = changeRequest.getCaseRoleId();
         CaseData caseDataNew = caseDetailsNew.getCaseData();
-        String partyName = NocNotificationHelper.getRespondentNameForNewSolicitor(changeRequest, caseDataNew);
         CaseData caseDataPrevious = caseDetailsPrevious.getCaseData();
-        if (!isClaimantNonSystemUser(caseDataPrevious)) {
-            sendClaimantEmail(caseDetailsPrevious, caseDetailsNew, partyName);
-        }
+        String partyName;
 
-        if (changeRequest.getOrganisationToRemove() != null) {
-            String previousOrgId = changeRequest.getOrganisationToRemove().getOrganisationID();
-            sendEmailToOldOrgAdmin(previousOrgId, caseDataPrevious);
+        if (caseRoleId.getValue().getCode().equals(ClaimantSolicitorRole.CLAIMANTSOLICITOR.getCaseRoleLabel())) {
+            // send claimant noc change email
+            partyName = NotificationHelper.getNameForClaimant(caseDataPrevious);
```

Comment:
Should this not just be the claimant's name? In the instance of a rep coming onto the case for the very first time, this logic is fine but representation can change and say if a new rep is assigned, based on the logic in NotificationHelper.getNameForClaimant, it would pull the old reps name out and they would appear as the party_name on the email

```suggestion
            partyName = caseDataPrevious.getClaimant();
```

<img width="724" height="565" alt="Image" src="https://github.com/user-attachments/assets/a478b3a3-a60a-41cb-8d20-b7523bfc921b" />

Reply:
resolved [3aa5621](https://github.com/hmcts/et-ccd-callbacks/pull/2696/commits/3aa562179eba0c9b82ad2f174207da9df79d394e)